pandas/tests/io/test_excel.py

```diff
@@ -1058,51 +1008,34 @@ def test_read_excel_squeeze(self):
         tm.assert_series_equal(actual, expected)
 
 
-class TestXlsReaderTests(XlrdTests):
-    ext = '.xls'
-    engine_name = 'xlrd'
-    check_skip = staticmethod(_skip_if_no_xlrd)
-
-
-class TestXlsxReaderTests(XlrdTests):
-    ext = '.xlsx'
-    engine_name = 'xlrd'
-    check_skip = staticmethod(_skip_if_no_xlrd)
+class _WriterBase(SharedItems):
```

Comment:
The original `ExcelWriterBase` class had an explicit setup / teardown method to set / reset options before each test case. Because I parametrized that test the setup / teardown paradigm did not have visibility into the actual fixtures being passes, so setting up an autoused fixture here to wrap each test case was the best option to inspect the parameters and set / reset the appropriate excel options before and after test execution.

I made this it's own subclass because the three classes that test specific features of `openpyxl`, `xlsxwriter` and `xlwt` still benefit from having this fixture, although it does require them to provide some extra metadata (namely the `merge_cells` fixture) that they don't use.

If this is too wonky, I think an alternate approach would be to re-wire the tests to specifically pass the engine fixture as a keyword argument to their read functions (didn't explore that option in detail so could be wrong)